pkg/datapath/linux/node_ids.go

```diff
@@ -77,8 +72,28 @@ func (n *linuxNodeHandler) GetNodeIP(nodeID uint16) string {
 	return n.nodeIPsByIDs[nodeID]
 }
 
+func (n *linuxNodeHandler) GetNodeID(nodeIP net.IP) (uint16, bool) {
+	n.mutex.Lock()
+	defer n.mutex.Unlock()
+
+	return n.getNodeIDForIP(nodeIP)
+}
+
+func (n *linuxNodeHandler) getNodeIDForIP(nodeIP net.IP) (uint16, bool) {
+	localNode := node.GetIPv4()
+	if localNode.Equal(nodeIP) {
+		return 0, true
```

Comment:
why are we returning true if we are not suppose to allocate a node ID for the local node?

Reply:
@jibi `0` is still the node id for the local node (and therefore has its meaning for users of this).